### DISCUSSION TITLE
Add-ons?

### DISCUSSION THREAD

**Discussion by samusishere** [BimmerGestalt/AAIdrive#421] [Q&A] [answered]
I saw recently that a new feature in the nightly build called add-ons was recently added in AAID. after looking around stumbled upon the development page for additional app compatibility and saw that screen mirroring is one of the apps being worked on, how would I enable this?

**Comment by hufman** [comment, marked as answer]
Indeed! Addons support is one of the new features being developed for the next version, to help allow other developers to easily build their own apps for the car. That repo contains several ideas and example implementations to help get started.

To install the pre-built Addons, you'll look in the [Commit Log](https://github.com/BimmerGestalt/IDriveConnectAddons/commits/main) to see the green checkmarks next to each commit. These will take you to the [Github CI](https://github.com/BimmerGestalt/IDriveConnectAddons/actions) for that commit, and you'll find a Build Artifact attached. This Build Artifact will contain the APKs that were compiled as of that version of the code, which you can install to add the addon to the app.

But remember, these are only simple samples! I welcome contributiona to them, but I can't devote very much time for handling feature requests.

**Reply by hufman** [reply]
I added a slight improvement, each Addon is split out to a separate APK on [this page](https://github.com/BimmerGestalt/IDriveConnectAddons/releases/tag/latest_build) instead of needing to download the zip of all of them from the Github Actions artifacts.

**Comment by bbiederm73** [comment]
Hi!
I don't get addons to work. Everything else works fine.
I am using version 1.3 and am trying the screen_mirror and cds_gauges addons. 
All three .APKs are installed separately. I don't get any icon on my android for the addons, just for AAIdrive.
"So far so good".

Now my problem:
In the readme.md it says "Apps which act as AAIdrive Addons are discovered and presented in the Addons tab". I can't find the addon tab. Neither in AAIdrive nor on my iDrive6 system. Where should I find that tab?

Did i do something wrong or did i misunderstand something ?!?

Thanks in advance
bbiederm73

**Reply by hufman** [reply]
Good morning! Addon support will be one of the big features for version 1.3.1, and is currently only in the Nightly Build. Please install [the nightly build](https://bimmergestalt.s3.amazonaws.com/aaidrive/builds/androidautoidrive-latest-main-nomap-sentry-release.apk) which will provide the Addons tab for you.

**Reply by bbiederm73** [reply]
Thanks alot! It is working now!
